Plugins/SequencePlugin/Source/SequencePlugin/Private/Types/ERC1155.cpp

```diff
@@ -0,0 +1,162 @@
+#include "Types/ERC1155.h"
+#include "ABI/ABI.h"
+
+FRawTransaction UERC1155::MakeGrantRoleTransaction(const FString& role, const FString& ToAddress)
+{
+	FString FunctionSignature = "grantRole(bytes32,address)";
+
+	FString Role = role;
+
+	TArray<uint8> ProofBytes;
+	ProofBytes.SetNumUninitialized(32);
+	FMemory::Memcpy(ProofBytes.GetData(), TCHAR_TO_UTF8(*Role), FMath::Min(Role.Len(), 32));
+
+	TFixedABIData ABIProof(ProofBytes);
+
+	FString WalletAddress = ToAddress;
+	FString WalletAddressNoPrefix = WalletAddress.Mid(2, WalletAddress.Len());
+	FAddress WalletAddressBytes = FAddress::From(WalletAddressNoPrefix);
+
+	TFixedABIData ABIAccount = ABI::Address(WalletAddressBytes);
+
+	TArray<ABIEncodeable*> Arr;
+
+	Arr.Add(&ABIProof);
+	Arr.Add(&ABIAccount);
+
+	FUnsizedData EncodedData = ABI::Encode(FunctionSignature, Arr);
+
+	FRawTransaction T;
+	T.data = "0x" + EncodedData.ToHex();
+	T.to = ContractAddress;
+	T.value = "0";
+
+	return T;
+}
+
+FRawTransaction UERC1155::MakeMintTransaction(const FString& ToAddress, const int32 TokenId, const int32 Amount)
+{
+	FString FunctionSignature = "mint(address,uint256,bytes)";
```

Comment:
I think this should be mint(address,uint256,uint256,bytes)

Reply:
updated